swarm/storage/feed/request_test.go

```diff
@@ -271,42 +212,30 @@ func TestReverse(t *testing.T) {
 	// generate a chunk key for this request
 	key := request.Addr()
 
-	if err = request.Sign(signer); err != nil {
-		t.Fatal(err)
-	}
+	err := request.Sign(signer)
+	t.Ok(err)
 
 	chunk, err := request.toChunk()
-	if err != nil {
-		t.Fatal(err)
-	}
+	t.Ok(err)
 
 	// check that we can recover the owner account from the update chunk's signature
 	var checkUpdate Request
-	if err := checkUpdate.fromChunk(chunk); err != nil {
-		t.Fatal(err)
-	}
+	err = checkUpdate.fromChunk(chunk)
+	t.Ok(err)
+
 	checkdigest, err := checkUpdate.GetDigest()
-	if err != nil {
-		t.Fatal(err)
-	}
-	recoveredAddr, err := getUserAddr(checkdigest, *checkUpdate.Signature)
-	if err != nil {
-		t.Fatalf("Retrieve address from signature fail: %v", err)
-	}
+	t.Ok(err)
+
+	recoveredAddr, err := getUserAddr(checkdigest, *checkUpdate.Signature) //Retrieve address from signature
+	t.Ok(err)
+
 	originalAddr := crypto.PubkeyToAddress(signer.PrivKey.PublicKey)
 
 	// check that the metadata retrieved from the chunk matches what we gave it
-	if recoveredAddr != originalAddr {
-		t.Fatalf("addresses dont match: %x != %x", originalAddr, recoveredAddr)
-	}
+	t.Assert(recoveredAddr == originalAddr, "addresses dont match: %x != %x", originalAddr, recoveredAddr)
+	t.Equals(chunk.Address()[:], key[:])
```

Comment:
One of the gotchas of the `t.Equals` that we would have to be very careful with:

```
	a := 1
	b := 1

	t.Equals(a, b)
	t.Equals(&a, &b)

	if a != b {
		t.Fatal("a != b")
	}

	if &a != &b {
		t.Fatal("&a != &b")
	}
```

This is one of the reasons I don't like frameworks like `testify`.

Reply:
@jpeletier as explained during the call, types are not always compared as you would expect.